src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+            ResultSet resultSet = allMfStatement.executeQuery();
+
+            while (resultSet.next()) {
+                allManufacturers.add(getManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't inquiry [manufacturers] SQL database", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getRequest = "SELECT * FROM manufacturers"
+                + " WHERE id = ? AND is_deleted = FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement selectMfByIdStatement =
+                        connection
+                             .prepareStatement(getRequest)) {
+            selectMfByIdStatement.setLong(1, id);
+            ResultSet resultSet = selectMfByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't obtain data from [manufacturers] "
+                    + "SQL database for id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?,"
+                + " country = ? WHERE id = ? AND is_deleted = FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateMfStatement =
+                        connection.prepareStatement(updateRequest)) {
+
+            updateMfStatement.setString(1, manufacturer.getName());
+            updateMfStatement.setString(2, manufacturer.getCountry());
+            updateMfStatement.setLong(3, manufacturer.getId());
+            updateMfStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update entry from [manufacturers] SQL "
+                    + "database for manufacturer" + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET"
+                + " is_deleted = TRUE WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement =
+                        connection.prepareStatement(deleteRequest)) {
+            deleteManufacturerStatement.setLong(1, id);
+
+            return deleteManufacturerStatement.executeUpdate() >= 1;
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete entry from [manufacturers] "
+                    + "SQL database for id: " + id, e);
+        }
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
a bit more fitting name
```suggestion
    private Manufacturer parseManufacturer(ResultSet resultSet) throws SQLException {
```